Cython/Compiler/Symtab.py

```diff
@@ -42,6 +42,22 @@ def c_safe_identifier(cname):
         cname = Naming.pyrex_prefix + cname
     return cname
 
+def punycodify_name(cname):
+    # modified from  PEP489
+    try:
+        cname = cname.encode('ascii')
+    except UnicodeEncodeError:
+        cname = cname.encode('punycode').replace(b'-', b'_')
+        if cname.startswith(Naming.pyrex_prefix.encode('ascii')):
```

Comment:
I would rather like to see punicode identifiers use a different prefix. For example, we use `__pyx_v_` for variables. Why not use `__pyx_vp_` for punicode variable names?

Reply:
The line after this snippet does that, so a variable becomes `__pyx_Uv_`; a function becomes `__pyx_Uf`. It does that just by inserting the `U` rather than adding more prefixes `Naming.py`.

(Where that might cause problems is on unicode attributes in unicode classes - only one `U` gets inserted so it isn't clear which part of the identifier is unicode. I think if you're really unlucky you could end up with a name collision, but it's pretty unlikely)

Reply:
Then, let's make that an official `pyunicode_identifier_prefix` in `Naming.py`, so that it's visible where the prefix comes from.